tactics/btermdn.ml

```diff
@@ -89,7 +89,12 @@ let evaluable_named id env ts =
   (try Environ.evaluable_named id env with Not_found -> true) &&
   (match ts with None -> true | Some ts -> TransparentState.is_transparent_variable ts id)
 
+(* The pattern view functions below try to overapproximate βι-neutral terms up
+   to η-conversion. Some historical design choices are still incorrect w.r.t. to
+   this specification. TODO: try to make them follow the spec. *)
```

Comment:
Since you are already adding comments, could you elaborate a bit on this specification? I am not sure I really understand `βι-neutral`. Are those neutral terms (in the traditional sense) that also do not create new iota reduction steps when composed via application?
It might also be worth expanding on the `η-conversion` bit since that depends on the optional transparency state IIUC.

Reply:
> Are those neutral terms (in the traditional sense)

Yes, neutral means that it's normal and it won't create a redex when substituted inside a context (not only β via application, but also ι when substituted inside a case). This is the reason we need to be clever w.r.t. to η-expansion since it might turn an actual neutral into a non-neutral. Also note that constructors are not neutral but we consider them here nonetheless for pattern recognition.

> that depends on the optional transparency state

Does it? We perform η-contraction regardless of the ts. Lambda-abstractions are not classified the same whether we're discriminated or not, but it should be uncommon to hit a difference and it shouldn't affect the η-quotient anyways. In any case I am going to try to get those behaviours even closer, so that in the end the only different between them is that in undiscriminated mode all references should be considered transparent.

Reply:
> Does it? We perform η-contraction regardless of the ts. 

Oh, of course. I am not sure why but even after spending half a minute looking for the right key combination to write `η` I still read it as `δ`. Disregard that part of my comment.